src/main/java/org/openstreetmap/atlas/checks/maproulette/MapRouletteConfiguration.java

```diff
@@ -51,12 +55,85 @@ public static MapRouletteConfiguration parse(final String configuration)
         return null;
     }
 
+    /**
+     * Parses a map roulette configuration object from a string that follows the structure
+     * [SERVER]:[PORT]:[PROJECT_NAME]:[API_KEY] and a project-specific configuration object. Note
+     * that the project configuration object also contains a project name -- if the two differ, a
+     * warning will be logged and the project name from the string will be used.
+     * 
+     * @param configuration
+     *            The configuration string to parse.
+     * @param projectConfiguration
+     *            A ProjectConfiguration defining the project that this connection will connect to
+     *            on map roulette
+     * @return A valid Map Roulette configuration object, null returned if the configuration string
+     *         is not valid.
+     */
+    public static MapRouletteConfiguration parseWithProject(final String configuration,
```

Comment:
Do we need an additional function name here? We could just overload the `parse` function

Reply:
I looked at it again, and after making the changes that @Bentleysb requested this isn't really needed at all, so I removed it altogether